pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -79,6 +78,14 @@ func (c *Reconciler) createOrUpdateAffinityAssistantsPerAABehavior(ctx context.C
 		}
 	}
 
+	// create PVCs from PipelineRun's VolumeClaimTemplate when aaBehavior is AffinityAssistantPerWorkspace or AffinityAssistantDisabled before creating
+	// affinity assistant so that the OwnerReference of the PVCs are the pipelineruns, which is used to achieve PVC auto deletion at PipelineRun deletion time
+	if (aaBehavior == aa.AffinityAssistantPerWorkspace || aaBehavior == aa.AffinityAssistantDisabled) && pr.HasVolumeClaimTemplate() {
+		if err := c.pvcHandler.CreatePVCsForWorkspacesWithoutAffinityAssistant(ctx, pr.Spec.Workspaces, *kmeta.NewControllerRef(pr), pr.Namespace); err != nil {
```

Comment:
CreatePVCsForWorkspacesWithoutAffinityAssistant will fail if it tries to create PVCs that already exist, but createOrUpdateAffinityAssistantsAndPVCsPerAABehavior runs on every reconcile loop. Will this work past the first reconcile loop?

Reply:
Also, we need to update the docstring of CreatePVCsForWorkspacesWithoutAffinityAssistant https://github.com/tektoncd/pipeline/blob/main/pkg/reconciler/volumeclaim/pvchandler.go#L54-L59

Reply:
No it will be only executed once, as it is guarded by `pipelineRunFacts.State.IsBeforeFirstTaskRun()`. The `CreatePVCsForWorkspacesWithoutAffinityAssistant ` also checks the existence of the PVCs